File: leetcode/965-单值二叉树/965-单值二叉树.js
/**
 * Definition for a binary tree node.
 *  */
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

/**
 * @param {TreeNode} root
 * @return {boolean}
 */

// 965-单值二叉树
var isUnivalTree = function (root) {
    if (!root) {
        return true;
    }
    if (root.left) {
        if (root.left.val !== root.val || !isUnivalTree(root.left)) {
            return false;
        }
    }

    if (root.right) {
        if (root.right.val !== root.val || !isUnivalTree(root.right)) {
            return false;
        }
    }
    return true;
};

File: leetcode/965-单值二叉树/965-单值二叉树.py


# 965-单值二叉树
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    
    def isUnivalTree(self, root: Optional[TreeNode]) -> bool:  
        if root is None:
            return True
        if root.left:
            if root.val != root.left.val or not self.isUnivalTree(root.left):
                return False
        if root.right:
            if root.val != root.right.val or not self.isUnivalTree(root.right):
                return False

        return True
        